### PR TITLE
Unit tests for some auxiliary classes

### PR DESCRIPTION
**Pull request by eryeer**
Here are some unit tests for some auxiliary classes.

PS: I removed the `UT_NefFile.cs` because it is duplicated with `SmartContract/UT_NefFile.cs`

### CODE REVIEW

File: neo.UnitTests/IO/UT_ByteArrayComparer.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Neo.IO;
+
+namespace Neo.UnitTests.IO
+{
+    [TestClass]
+    public class UT_ByteArrayComparer
```
Comment (vncoelho) [inline neo.UnitTests/IO/UT_ByteArrayComparer.cs]: Which calls usually uses this `ByteArrayComparer`?

File: neo.UnitTests/UT_Helper.cs [hunk 2]
```diff
@@ -38,5 +43,271 @@ public void ToScriptHash()
...
+
+            var list3 = new List<Woo>();
+            list3.WeightedFilter(0.3, 0.6, p => p.Value, (p, w) => new Result
+            {
+                Info = p,
+                Weight = w
+            }).WeightedAverage(p => p.Weight, p => p.Weight).Should().Be(0);
+
+            list2 = null;
+            Action action6 = () => list2.WeightedFilter(0.3, 0.4, p => p.Value, (p, w) => new Result
```
Comment (vncoelho) [inline neo.UnitTests/UT_Helper.cs]: Sounds good. Could these actions checks be generalized into a function for becoming a little bit more compact?

**Comment by vncoelho** [comment]
@eryeer, I do not see the coverage report. Do you know why?

**Comment by eryeer** [comment]
> @eryeer, I do not see the coverage report.
> It is behaving strange sometimes.

Yes, it's strange. Let's check what happen to the code coverage.

File: neo.UnitTests/IO/UT_ByteArrayComparer.cs (hunk 1 above)
Comment (eryeer) [reply neo.UnitTests/IO/UT_ByteArrayComparer.cs]: It is used by `Find()` in `DataCache.cs` to compare byte arrays either from cache or storage to provide a sequential return value.

**Comment by shargon** [comment]
Where is the coverage report?

**Comment by eryeer** [comment]
> Where is the coverage report?

Seems all recent PR don't have the coverage report. Don't know the happend.

**Comment by vncoelho** [comment]
It is strange, @eryeer  and @shargon, 

image


It is not recognizing some commits as well,

image

image


Some do not have the "CI Passed " flag as well.
Other ones, when we click on "Processing" tell that private repo message and then turn to "error"

**Pull request merged**